### PR TITLE
Feature/game over

### PR DESCRIPTION
**Pull request by KyleWong2510**
## Pull request checklist
Please check if your PR fulfills the following requirements:
- [ ] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type
 
 
Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Testing
- [ ] Other (please describe): 

## What is the current behavior?

- No way to move to new questions
- No counters in header
- No end of game

## What is the new behavior?

- Create AnswerModal and add all of its functionality
- Increment/decrement score
- Decrement lives
- Increment currentQuestion counter
- Start GameOver component

## Does this introduce a breaking change?
- [ ] Yes
- [x] No

## Other information

Need to figure out how to render GameOver when the game is completed without losing all lives.

### CODE REVIEW

File: src/Chalkboard/Chalkboard.js [hunk 1]
```diff
@@ -3,35 +3,137 @@ import { connect } from 'react-redux'
...
+  decrementLives,
+  incrementScore,
+  decrementScore 
+} from '../actions'
 
 class Chalkboard extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state={
+      question: this.props.questions[this.props.currentRound][this.props.currentQuestion],
```
Comment (lilianaweimer) [inline src/Chalkboard/Chalkboard.js:22]: Hooray for developer empathy! Love that this makes things more readable

File: src/Chalkboard/Chalkboard.js [hunk 2]
```diff
@@ -3,35 +3,137 @@ import { connect } from 'react-redux'
...
+    this.state.question.difficulty === 'hard' && this.props.incrementScore(150)
+  }
+
+  removePoints = () => {
+    this.state.question.difficulty === 'easy' && this.props.decrementScore(25)
+    this.state.question.difficulty === 'medium' && this.props.decrementScore(50)
+    this.state.question.difficulty === 'hard' && this.props.decrementScore(75)
+  }
+  
+  incrementQuestion = async () => {
```
Comment (lilianaweimer) [inline src/Chalkboard/Chalkboard.js:63]: Awesome use of async/await here!

File: src/Chalkboard/Chalkboard.js [hunk 3]
```diff
@@ -3,35 +3,137 @@ import { connect } from 'react-redux'
...
+    }
+  }
+
+  addPoints = () => {
+    this.state.question.difficulty === 'easy' && this.props.incrementScore(50)
+    this.state.question.difficulty === 'medium' && this.props.incrementScore(100)
+    this.state.question.difficulty === 'hard' && this.props.incrementScore(150)
+  }
+
+  removePoints = () => {
```
Comment (Billwilke42) [inline src/Chalkboard/Chalkboard.js:57]: Nice work on this

File: src/Chalkboard/Chalkboard.js [hunk 4]
```diff
@@ -3,35 +3,137 @@ import { connect } from 'react-redux'
...
+        />
+        <section className='body'>
+          <section className='chalkboard'>
+            <Question question={this.state.question.question} />
+            <Answers 
+              question={this.state.question} 
+              checkAnswer={this.checkAnswer}
+            />
+            {this.state.isCorrect && 
+              <AnswerModal
```
Comment (Billwilke42) [inline src/Chalkboard/Chalkboard.js:94]: Answer Modal was a great idea!

File: src/actions/index.js [hunk 5]
```diff
@@ -3,23 +3,30 @@ export const setQuestions = questions => ({
...
+export const resetCurrentQuestion = () => ({
+  type: 'RESET_CURRENT_QUESTION',
+})
+
+export const incrementScore = score => ({
+  type: 'INCREMENT_SCORE',
+  score
+})
+
+export const decrementScore = score => ({
```
Comment (Billwilke42) [inline src/actions/index.js:23]: Nice work on getting the score to decrement on wrong answers

File: src/reducers/setScore.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+const initialState = 0
+
+export const setScore = (state = initialState, action) => {
```
Comment (Billwilke42) [inline src/reducers/setScore.js:3]: Nice work making this dynamic

**Pull request merged**